packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -175,11 +176,41 @@ function mountElement(vnode: VElement, parent: ParentNode, anchor: Node | null)
         data: { svg },
     } = vnode;
 
+    if (vnode.isStatic) {
+        if (vnode.elm) {
+            // @todo: what to do about the owner for synthetic shadow and fallbackElmHook
+
+            // @todo: cloned subtree is not associated with the proper shadow. The restrictions and styles will be
+            //        ok.
+            //        maybe modify synthetic shadow and get the owner from ancestors?
+            vnode.elm = vnode.elm.cloneNode(true) as Element;
+
+            linkNodeToShadow(vnode.elm, owner, true);
+
+            if (process.env.NODE_ENV !== 'production') {
+                // @todo: take out the restrictions patching
+                // @todo: should we traverse the tree?
+                fallbackElmHook(vnode.elm, vnode);
+            }
+
+            insertNode(vnode.elm, parent, anchor);
+            return;
+        }
+
+        // @todo: hackalert: for now, let's mark all children as static and set the owner
+        const ch = vnode.children;
+        for (let i = 0, n = ch.length; i < n; i++) {
+            ch[i]!.isStatic = true;
+            ch[i]!.owner = owner;
+        }
```

Comment:
Is this a hack because it could be done in the `tmpl` function?

Reply:
well, it may not be a hack 😄. It can not be done in the template function because that will be executed in every rehydration, while here only executes just one time per component (!== component instance), when we need to create the element.

I marked it as a hack because I didn't give too much thought then, maybe there's a better way of achieving this.

The key idea is that we need to go through the regular mounting mechanism to mount the hoisting tree (just one time per module!) but only the root vnode is marked as static and has an owner, it may be the case that contains other vnodes, and we need to set the owner in them.

Reply:
Sorry, I didn't mean to say "in the `tmpl` function," I meant "can the compiled code do this?" Or can we do it in `setOwner`?

Reply:
the compiled code can not do this, because there's no owner at the time that the vnode is created.

if we do it in `setOwner` we would be doing it for every re-hydration, vs just one time per module.

Reply:
OK got it, thank you!